### PR TITLE
Turn on StrictNullChecks for rush & rush-lib

### PR DESCRIPTION
**Pull request by nickpape-msft**

### CODE REVIEW

File: apps/rush-lib/src/data/PinnedVersionsConfiguration.ts [hunk 1]
```diff
@@ -79,11 +80,13 @@ export class PinnedVersionsConfiguration {
...
    * Preferred to use PinnedVersionsConfiguration.loadFromFile()
    */
-  private constructor(pinnedVersionJson: IPinnedVersionsJson, private _filename: string) {
+  private constructor(pinnedVersionJson: IPinnedVersionsJson | undefined, private _filename: string) {
     this._data = new Map<string, string>();
-    Object.keys(pinnedVersionJson || {}).forEach((dep: string) => {
-      this.set(dep, pinnedVersionJson[dep]);
-    });
+    if (pinnedVersionJson) {
+      Object.keys(pinnedVersionJson).forEach((dep: string) => {
```
Comment (pgonzal) [inline apps/rush-lib/src/data/PinnedVersionsConfiguration.ts]: >Object.keys(pinnedVersionJson).forEach((dep: string) => { [](start = 6, length = 57)

Better to do this:

```
for (const dep of Object.keys(pinnedVersionJson)) {
}
```

File: apps/rush-lib/src/data/PinnedVersionsConfiguration.ts [hunk 2]
```diff
@@ -79,11 +80,13 @@ export class PinnedVersionsConfiguration {
...
+        this.set(dep, pinnedVersionJson[dep]);
```
Comment (pgonzal) [inline apps/rush-lib/src/data/PinnedVersionsConfiguration.ts]: >dep [](start = 40, length = 3)

our coding conventions require full English words

File: apps/rush-lib/src/data/CommandLineConfiguration.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
```
Comment (pgonzal) [inline apps/rush-lib/src/data/CommandLineConfiguration.ts]: >// [](start = 0, length = 2)

Where did this file come from?  What does it have to do with strict null checks?

File: apps/rush-lib/src/utilities/VersionControl.ts [hunk 4]
```diff
@@ -7,9 +7,9 @@ import * as child_process from 'child_process';
  * @public
  */
 export default class VersionControl {
-  public static getChangedFolders(targetBranch?: string): string[] {
+  public static getChangedFolders(targetBranch?: string): Array<string | undefined> | undefined {
```
Comment (pgonzal) [inline apps/rush-lib/src/utilities/VersionControl.ts]: >undefined [](start = 73, length = 9)

Why are we doing this?  Seems like a bad design

Comment (nickpape-msft) [reply apps/rush-lib/src/utilities/VersionControl.ts]: I thought so too, but I wasn't trying to change any behavior in this PR.

---
In reply to: [142773890](https://github.com/Microsoft/web-build-tools/pull/385#discussion_r142773890) [](ancestors = 142773890)

File: apps/rush-lib/src/data/CommandLineConfiguration.ts (hunk 3 above)
Comment (nickpape-msft) [reply apps/rush-lib/src/data/CommandLineConfiguration.ts]: It was accidently added, is a part of the next PR I was working on

---
In reply to: [142773840](https://github.com/Microsoft/web-build-tools/pull/385#discussion_r142773840) [](ancestors = 142773840)

**Pull request merged**